optuna/storages/rdb/storage.py

```diff
@@ -912,61 +914,169 @@ def get_trial_number_from_id(self, trial_id):
-        params = models.TrialParamModel.where_trial(trial, session)
-        values = models.TrialValueModel.where_trial(trial, session)
-        user_attributes = models.TrialUserAttributeModel.where_trial(trial, session)
-        system_attributes = models.TrialSystemAttributeModel.where_trial(trial, session)
+        return copy.deepcopy(frozen_trial)
 
-        frozen_trial = self._merge_trials_orm(
-            [trial], params, values, user_attributes, system_attributes
-        )[0]
+    def get_all_trials(self, study_id, deepcopy=True):
+        # type: (int, bool) -> List[FrozenTrial]
 
-        self._finished_trials_cache.cache_trial_if_finished(frozen_trial)
+        trials = self._get_uncached_trials(study_id, set())
 
-        # Terminate transaction explicitly to avoid connection timeout during transaction.
-        self._commit(session)
+        return copy.deepcopy(trials) if deepcopy else trials
 
-        return copy.deepcopy(frozen_trial) if deepcopy else frozen_trial
+    def _get_uncached_trials(self, study_id: int, cached_trial_ids: Set[int]) -> List[FrozenTrial]:
 
-    def get_all_trials(self, study_id, deepcopy=True):
-        # type: (int, bool) -> List[FrozenTrial]
+        session = self.scoped_session()
 
-        if not self._finished_trials_cache.is_empty():
-            trial_ids = self._get_all_trial_ids(study_id)
-            # Check if no more than 5 trials are missing from the cache.
-            # This prevents a single item in the cache from resulting in O(N) database lookups when
-            # the bulk fetch below would be significantly faster.
-            if (
-                sum(
-                    not self._finished_trials_cache.get_cached_trial(trial_id)
-                    for trial_id in trial_ids
-                )
-                < 5
-            ):
-                trials = [self._get_and_cache_trial(trial_id, deepcopy) for trial_id in trial_ids]
-                return trials
+        # Ensure that the study exists.
+        models.StudyModel.find_or_raise_by_id(study_id, session)
 
-        # Cache is either empty or missing enough elements that a bulk fetch is better.
-        trials = self._get_all_trials_without_cache(study_id)
-        for trial in trials:
-            self._finished_trials_cache.cache_trial_if_finished(trial)
+        trial_models = (
+            session.query(
+                models.TrialModel.trial_id,
+                models.TrialModel.state,
+                models.TrialModel.value,
+                models.TrialModel.datetime_start,
+                models.TrialModel.datetime_complete,
+                models.TrialModel.number,
+            )
+            .filter(
+                ~models.TrialModel.trial_id.in_(cached_trial_ids),
+                models.TrialModel.study_id == study_id,
+            )
+            .all()
+        )
+        # On single-worker training, ``trial_models`` is always an empty list.
```

Comment:
Let me suggest the change below because "optimization" sounds a bit more Optuna-like term to me.
```suggestion
        # On single-worker optimization, ``trial_models`` is always an empty list.
```